### PR TITLE
enum smoke test

### PR DESCRIPTION
**Pull request by viacheslav-rostovtsev**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:

- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/google-cloud-ruby/issues) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea.
- [ ] Follow the instructions in [CONTRIBUTING](CONTRIBUTING.md). Most importantly, ensure the tests and linter pass by running `bundle exec rake ci` in the gem subdirectory.
- [ ] Update code documentation if necessary.

closes: #<issue_number_goes_here>

### CODE REVIEW

File: google-cloud-compute-v1/test/google/cloud/compute/v1/misc_test.rb [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    # UNDEFINED_STATE = 0;
+    # ACTIVE = 314733318;
+    # DELETED = 120962041;
+    # DEPRECATED = 463360435;
+    # OBSOLETE = 66532761;
+    # (see https://raw.githubusercontent.com/googleapis/googleapis-discovery/master/google/cloud/compute/v1/compute.proto)
+
+    unknown_enum_json = deprecation_status_json.gsub "OBSOLETE", "THIS_VALUE_DOES_NOT_EXIST"
+    ds_decoded = ::Google::Cloud::Compute::V1::DeprecationStatus.decode_json unknown_enum_json, ignore_unknown_fields: true
+    assert_equal "THIS_VALUE_DOES_NOT_EXIST", ds_decoded.state
```
Comment (dazuma) [inline google-cloud-compute-v1/test/google/cloud/compute/v1/misc_test.rb:29]: Are we expecting `UNDEFINED_STATE` here?

File: google-cloud-compute-v1/test/google/cloud/compute/v1/misc_test.rb [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+  def test_parse_numeric_enum_value
+    deprecation_status_json = Google::Cloud::Compute::V1::DeprecationStatus.new(
+      deleted: "foo",
+      state: :OBSOLETE).to_json
+
+    ds_decoded = ::Google::Cloud::Compute::V1::DeprecationStatus.decode_json deprecation_status_json, ignore_unknown_fields: true
+    assert_equal "OBSOLETE", ds_decoded.state
+    
+    unknown_enum_json = deprecation_status_json.gsub "OBSOLETE", "57"
```
Comment (dazuma) [inline google-cloud-compute-v1/test/google/cloud/compute/v1/misc_test.rb:40]: Is this supposed to be the _string_ `"57"` in the json, or the number `57`? Currently I think this turns it into the string, which basically makes this equivalent to the previous test (the string value `"57"` does not exist).

**Review by dazuma — COMMENTED** [review]
A couple questions

File: google-cloud-compute-v1/test/google/cloud/compute/v1/misc_test.rb (hunk 2 above)
Comment (viacheslav-rostovtsev) [reply google-cloud-compute-v1/test/google/cloud/compute/v1/misc_test.rb]: I've updated the test to use a number, which (expectedly) raises an error.
```
 1) Error:                                                                                                                                                                                                                                                                                                                                                                                 
MiscTest#test_parse_numeric_enum_value:                                                                                                                                                                                                                                                                                                                                                     
Google::Protobuf::ParseError: Error occurred during parsing: Error parsing JSON @1:26: Expected string                                                                                                                                                                                                                                                                                      
    /home/virost/src/google-cloud-ruby/google-cloud-compute-v1/test/google/cloud/compute/v1/misc_test.rb:41:in `decode_json'                                                                                                                                                                                                                                                                
    /home/virost/src/google-cloud-ruby/google-cloud-compute-v1/test/google/cloud/compute/v1/misc_test.rb:41:in `test_parse_numeric_enum_value'                                                                                                                                                                                                                                              
```

File: google-cloud-compute-v1/test/google/cloud/compute/v1/misc_test.rb [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    
+    op_decoded = ::Google::Cloud::Compute::V1::Operation.decode_json operation_json, ignore_unknown_fields: true
+    assert_equal :PENDING, op_decoded.status
+
+    # Passing an unknown int works, with or without the `ignore_unknown_fields: true` option
+    int_enum_json = operation_json.gsub '"PENDING"', '42'
+    op_decoded = ::Google::Cloud::Compute::V1::Operation.decode_json int_enum_json, ignore_unknown_fields: true
+    assert_equal 42, op_decoded.status
+    op_decoded = ::Google::Cloud::Compute::V1::Operation.decode_json int_enum_json
+    assert_equal 42, op_decoded.status
```
Comment (vchudnov-g) [inline google-cloud-compute-v1/test/google/cloud/compute/v1/misc_test.rb:75]: Now calling `op_decoded.to_json` will have the number 42 in the JSON output, correct?

Comment (viacheslav-rostovtsev) [reply google-cloud-compute-v1/test/google/cloud/compute/v1/misc_test.rb:75]: yes. I've updated the test to check for this.

Comment (vchudnov-g) [reply google-cloud-compute-v1/test/google/cloud/compute/v1/misc_test.rb:75]: Cool, thanks. But does the assertion below check that the number is not in quotes? Or did you check that manually?

**Pull request closed without merging**